### DISCUSSION TITLE
Save request and response json details in variables

### DISCUSSION THREAD

**Discussion by Walrusv** [pactumjs/pactum#81] [Q&A] [answered]
Hello! 
I wonder if you can help me,
I would like to save request (and response) json in variables, 
I know about Inspect(), but this method just print in Terminal and it doesn't allow me to save it to a variable

This is my code 

```javascript
Given(/^Post Notificacion con atributos (.*), (.*), (.*)$/, async function(email, body, petition){
servicio.post('/telematics/v1/notification')
  .withHeaders({
    'Content-Type': 'application/json',
  })
  .withJson({ email: email,
    body: body,
    petition: petition
  })
  .inspect();
  await servicio.toss();
```

This is a example of request details printed in Terminal by the Inspect()
```javascript
  var requestDetail = {
    "url": "/telematics/v1/notification",
    "method": "POST",
    "headers": {
      "Content-Type": "application/json",
   },
    "path": "/telematics/v1/notification",
    "body": {
      "email": "rtrejo.zoe@gmail.com",
      "body": "test"
    },
    "timeout": 3000
  }

  
```
My goal is save in variables to attach in Cucumber Html Report,  How can i do?

```javascript
this.attach(JSON.stringify(requestDetail, null, 2));
this.attach(JSON.stringify(responseDetail, null, 2));
  
```

**Comment by leelaprasadv** [comment, marked as answer]
Hey @Walrusv !
Yes, `inspect()` prints the request, response onto terminal/console.
You can access the request and response details from spec object with `spec._request` and `spec._response`. In your case, it would be
```js
servicio._request // to get the request
servicio._response // to get the response
```
Just add these after the `toss()` and you might want to explicitly access parts from `_response` (like body, headers).

**Reply by Walrusv** [reply]
Thank you so much! @leelaprasadv  

**spec._reques**t- > Works perfectly! 
But " **spec._response**" prints a lot of information that I don't need or is irrelevant to response details

Example:

![image](https://user-images.githubusercontent.com/42990194/131175292-d6a2f512-5659-4180-833f-914c3a0c92ae.png)
![image](https://user-images.githubusercontent.com/42990194/131175329-c231e244-6987-4bcf-8439-6de44052b1d6.png)



Here is my code: 
```javascript
Given(/^Post Notificacion con atributos (.*), (.*), (.*)$/, async function(email, body, petition){
servicio.post('/telematics/v1/notification')
  .withHeaders({
    'Content-Type': 'application/json',
  })
  .withJson({ email: email,
    body: body,
    petition: petition
  });
  await servicio.toss();

//just for logs and validate
console.log(servicio._request); 
  console.log(servicio._response);
```


How can i get a correct response detail like the request one?

**Reply by leelaprasadv** [reply]
You can pick required details from the `_response` object. 
one easiest way to do it is with `lodash` npm package's `pick()`.

```js
 console.log(_.pick(servicio._response, ["headers", "statusCode", "json"]));
```

Or you could just do below without `lodash`

```js
let trimmedResponse = (({headers, statusCode, json}) => ({headers, statusCode, json}))(servicio._response);
console.log(trimmedResponse)
```

PS: `json` contains the response body

**Reply by Walrusv** [reply]
Thank you
I appreciate your help
It works!
